jakarta-eclipse/src/org/jakarta/jdt/BeanValidationDiagnosticsCollector.java

```diff
@@ -0,0 +1,306 @@
+
+		if (unit != null) {
+			IType[] alltypes;
+			IField[] allFields;
+			IAnnotation[] allFieldAnnotations;
+
+			try {
+				alltypes = unit.getAllTypes();
+				for (IType type : alltypes) {
+					allFields = type.getFields();
+					for(IField field: allFields) {
+
+
+						allFieldAnnotations = field.getAnnotations();
+						String fieldType = field.getTypeSignature();
+
+						for (IAnnotation annotation : allFieldAnnotations) {
+							if(SET_OF_ANNOTATIONS.contains(annotation.getElementName())) {
+								checkAnnoatationAllowedTypes(unit, diagnostics, fieldType, annotation);
+								
+								
+								if (Flags.isStatic(field.getFlags())) {
+									ISourceRange fieldAnnotationNameRange = JDTUtils.getNameRange(annotation);
+									Range fieldAnnotationrange = JDTUtils.toRange(unit, fieldAnnotationNameRange.getOffset(),
+											fieldAnnotationNameRange.getLength());
+									diagnostics.add(new Diagnostic(fieldAnnotationrange,
+											"Constraint Annotations are not allowed on static fields"));	
+								}
+							}
+							
+						}
+
+					}
+				}
+
+			} catch (JavaModelException e) {
+				Activator.logException("Cannot calculate diagnostics", e);
+			}
+		}
+
+	}
+
+	private void checkAnnoatationAllowedTypes(ICompilationUnit unit, List<Diagnostic> diagnostics, String fieldType,
+			IAnnotation annotation) throws JavaModelException {
+		
+		ISourceRange fieldAnnotationNameRange = JDTUtils.getNameRange(annotation);
+		Range fieldAnnotationrange = JDTUtils.toRange(unit, fieldAnnotationNameRange.getOffset(),
+				fieldAnnotationNameRange.getLength());
+		
+		if (annotation.getElementName().equals(ASSERT_FALSE)|| annotation.getElementName().equals(ASSERT_TRUE)) {
+
+			if (	!fieldType.equals(getSignatureFormatOfType(BOOLEAN)) &&
+					!fieldType.equals(Signature.SIG_BOOLEAN)) {
+				
+				diagnostics.add(new Diagnostic(fieldAnnotationrange,
+						"This annotation can only be used on boolean and Boolean type fields."));	
+			}
+		} else if (annotation.getElementName().equals(DECIMAL_MAX)|| annotation.getElementName().equals(DECIMAL_MIN) 
+				|| annotation.getElementName().equals(DIGITS) ) {
+
+			if (	!fieldType.equals(getSignatureFormatOfType(BIG_DECIMAL)) &&
+					!fieldType.equals(getSignatureFormatOfType(BIG_INTEGER)) &&
+					!fieldType.equals(getSignatureFormatOfType(CHAR_SEQUENCE)) &&
+					!fieldType.equals(getSignatureFormatOfType(BYTE)) &&
+					!fieldType.equals(getSignatureFormatOfType(SHORT)) &&
+					!fieldType.equals(getSignatureFormatOfType(INTEGER)) &&
+					!fieldType.equals(getSignatureFormatOfType(LONG)) &&
+					!fieldType.equals(Signature.SIG_BYTE) &&
+					!fieldType.equals(Signature.SIG_SHORT) &&
+					!fieldType.equals(Signature.SIG_INT) &&
+					!fieldType.equals(Signature.SIG_LONG)) {
+				
+				diagnostics.add(new Diagnostic(fieldAnnotationrange,
+						"This annotation can only be used on BigDecimal, BigInteger, CharSequence"
```

Comment:
Can we modify this diagnostic message (and those that are similar to it) to look like this, so that it is a bit easier to read:
```
"The @Digits annotation can only be used on: \n- BigDecimal \n- BigInteger \n- CharSequence"
						+ "\n- byte, short, int, long (and their respective wrappers) \n type fields."));	
```
![image](https://user-images.githubusercontent.com/26146482/100670208-8b163c80-332c-11eb-9755-5d0f2527f148.png)


For diagnostics that only check 2 or 3 types we can leave those messages all on one line.

Reply:
The @Past, @Future, ... annotations has so many allowed types that the diagnostic will run out of space so I kept them in a single line format as well for now.